pyscript.core/test/error.js

```diff
@@ -0,0 +1,77 @@
+// Shoelace
+const SHOELACE_BASE = 'https://cdn.jsdelivr.net/npm/@shoelace-style/shoelace@2.8.0/cdn';
```

Comment:
was this file supposed to be checked in or it sneaked in by mistake?

Reply:
Ahh... I see what you mean now. Probably sneaked in by accident.

Reply:
I think I've branched from that branch indeed ... silly me ... I'll fix that ASAP but it has nothing to do with the fix I am looking for.

Reply:
@antocuni @ntoll I've fixed bad rebase / branch issue ... and fixed our pre-hook config that was eating out `pyscript.py` and changing for no reasons `pyscript.js` ... that part is good now ... now:

> what about something like this ...

that doesn't work because we have multiple scripts running code. If we retrieve the *id* when `display()` is invoked it might be too late.

Previously `display` was a getter so that `from pyscript import display` was passing through that getter creating a unique `display` instrumented utility that by default would've used the right ID.

Of course this was assuming one might import `display` ASAP in the code but that's usually the case in every single test or example I could see.

Now that functionality is gone and reaching out anything too late won't likely work ever because there won't be a `currentScript` executing or reachable at all.

This is a bit of a bummer to me but I might have found a solution, although I am not sure I like it.

Reply:
@WebReflection ok... thanks for the additional context. Perhaps a function that retrieves the id on import and returns an inner function would work..? E.g. (off the top of my head) something like this:

```python
def display_with_id():
    """
    Calling this function creates a closure for the inner_display_fn which is what
    the module's display attribute is eventually bound to.
    """
    default_target = _internal.DISPLAY_TARGET  # retrieve the id when the function is called (on import)

    def inner_display_fn(*values, target=None, append=True):
        if target is None:
            target = default_target  # set target to the one retrieved when the outer function was called at import time
        if target is None:
            raise Exception(
                "Implicit target not allowed here. Please use display(..., target=...)"
            )
        for v in values:
            Element(target).write(v, append=append)

    return inner_display_fn

display = display_with_id()  # assigning display at import time causes display_with_id to retrieve the id and return the function inside the closure.
```

Reply:
Please check this... I'm improvising here. :-) TL;DR we're creating a closure context at import time when the ID is retrieved. The actual `display` is bound to the inner function that has access to the `default_target` gathered at import time.

Hope this makes sense.

Reply:
I think I've fixed it all and everything just works as expected now, thanks for [this hint](https://github.com/pyscript/pyscript/pull/1666#issuecomment-1702683448), it worked very well indeed!